internal/pkg/describe/app.go

```diff
@@ -0,0 +1,171 @@
+}
+
+// AppDescriber retrieves information about an application.
+type AppDescriber struct {
+	project string
+	app     string
+	env     string
+
+	ecsClient      ecsService
+	stackDescriber stackDescriber
+}
+
+// NewAppDescriber instantiates a new application.
+func NewAppDescriber(project, env, app string) (*AppDescriber, error) {
+	svc, err := store.New()
+	if err != nil {
+		return nil, fmt.Errorf("connect to store: %w", err)
+	}
+	meta, err := svc.GetApplication(project, app)
+	if err != nil {
+		return nil, fmt.Errorf("get application %s: %w", app, err)
+	}
+	environment, err := svc.GetEnvironment(project, env)
+	if err != nil {
+		return nil, fmt.Errorf("get environment %s: %w", env, err)
+	}
+	sess, err := session.NewProvider().FromRole(environment.ManagerRoleARN, environment.Region)
+	if err != nil {
+		return nil, err
+	}
+	d := NewStackDescriber(sess)
+	return &AppDescriber{
+		project: project,
+		app:     meta.Name,
+		env:     environment.Name,
+
+		ecsClient:      ecs.New(sess),
+		stackDescriber: d,
+	}, nil
+}
+
+// EnvVars returns the environment variables of the task definition.
+func (d *AppDescriber) EnvVars() (map[string]string, error) {
+	taskDefName := fmt.Sprintf("%s-%s-%s", d.project, d.env, d.app)
+	taskDefinition, err := d.ecsClient.TaskDefinition(taskDefName)
+	if err != nil {
+		return nil, err
+	}
+	envVars := taskDefinition.EnvironmentVariables()
+
+	return envVars, nil
+}
+
+// GetServiceArn returns the ECS service ARN of the application in an environment.
+func (d *AppDescriber) GetServiceArn() (*ecs.ServiceArn, error) {
+	appResources, err := d.stackDescriber.StackResources(stack.NameForApp(d.project, d.env, d.app))
+	if err != nil {
+		return nil, err
+	}
+	for _, appResource := range appResources {
+		if aws.StringValue(appResource.LogicalResourceId) == serviceLogicalID {
+			serviceArn := ecs.ServiceArn(aws.StringValue(appResource.PhysicalResourceId))
+			return &serviceArn, nil
+		}
+	}
+	return nil, fmt.Errorf("cannot find service arn in app stack resource")
+}
+
+// AppStackResources returns the filtered application stack resources created by CloudFormation.
+func (d *AppDescriber) AppStackResources() ([]*cloudformation.StackResource, error) {
+	appResources, err := d.stackDescriber.StackResources(stack.NameForApp(d.project, d.env, d.app))
+	if err != nil {
+		return nil, err
+	}
+	var resources []*cloudformation.StackResource
+	// See https://github.com/aws/amazon-ecs-cli-v2/issues/621
+	ignoredResources := map[string]bool{
+		rulePriorityFunction: true,
+		waitCondition:        true,
+		waitConditionHandle:  true,
+	}
+	for _, appResource := range appResources {
+		if ignoredResources[aws.StringValue(appResource.ResourceType)] {
+			continue
+		}
+		resources = append(resources, appResource)
+	}
+
+	return resources, nil
+}
+
+// EnvOutputs returns the output of the environment stack.
+func (d *AppDescriber) EnvOutputs() (map[string]string, error) {
```

Comment:
brainstorming 🤔 : We can keep it here I think for now, but maybe we should move this to @huanjani's EnvDescriber eventually? seems like it would make more sense there